[WIP] New builder - JDCloud (#7860)
hashicorp/packer: website/source/docs/builders/jdcloud.html.md

---
description: |
    The `jdcloud` Packer builder helps you to build instance images
    based on an existing image
layout: docs
page_title: 'JDCloud Image Builder'
sidebar_current: 'docs-builders-jdcloud'
---

# JDCloud Image Builder

Type: `jdcloud`

The `jdcloud` Packer builder helps you to build instance images
based on an existing image

## Configuration Reference

In order to build a JDCloud instance image, fullfill your configuration file. Necessary attributes
are given below: 

### Required-Parameters:

- `type` (string) - This parameter tells which cloud-service-provider you are using, in our case, use 'jdcloud'
- `image_id` (string) - New image is generated based on an old one, specify the base-image-id here. 

> - `access_key` (string) - Declare your identity , help us to know who you are. Alternatively you can set them as env-variable:`export JDCLOUD_ACCESS_KEY=xxx`

Reviewer: ```suggestion
- `access_key` (string) - Your JD Cloud access key. You may set this as an env-variable:`export JDCLOUD_ACCESS_KEY=xxx`
```

> - `secret_key` (string) - Same as access_key, write them this file, or set them as env-variable:`export JDCLOUD_SECRET_KEY=xxx`

Reviewer: ```suggestion
- `secret_key` (string) - Your JD Cloud secret key. You may also set this via env-variable:`export JDCLOUD_SECRET_KEY=xxx`
```

> - `region_id` (string) - Region of your instance, can be 'cn-north-1/cn-east-1' etc.

Reviewer: I would like to see a full list of available regions here, since they're hard-coded in the actual builder.

- `az` (string) - Exact availability zone of instance, 'cn-north-1c' for example
- `instance_name` (string) - Name your instance
- `instance_type` (string) - Class of your expected instance
- `image_name` (string) - Name the image you would like to create
- `ssh_username` (string) - Currently only `root` is supported 
- `communicator` (string) - Currently only `ssh` is supported. `winrm` will be added if required

> - `provisioners/inline` (string) - Commands, were written in an array form, where each command is a string, e.g it can be`apt-get update`

Reviewer: Provisioners aren't a builder option; you don't need to include this line. 

```suggestion
```

### Optional-Parameters

- `subnet_id` (string) - An instance is supposed to exists in an subnet, if not specified , we will create new one for you

> - `ssh_wait_timeout`(string) - 'Timeout' is introduced in trying to connect(ssh) to your instance

Reviewer: `ssh_wait_timeout` is a deprecated communicator option, and you don't need to document the communicator options here. Don't document anything in this builder except for your builder's options, so that we only have to maintain the communicator options in one place.
